example/echo-op/echo_op.cpp

```diff
@@ -13,13 +13,24 @@
 
 //[example_core_echo_op_1
 
+template<
+    class AsyncStream,
+    class CompletionToken>
+auto
+async_echo(AsyncStream& stream, CompletionToken&& token)
+
+//]
+    -> beast::async_return_type<CompletionToken, void(beast::error_code)>;
```

Comment:
I was originally going to say "missing semicolon," but now I see it. I presume you did the trailing return type because you're testing everything against a C++11 compiler?

Now that I see the deduced return type version live, I'm no longer sure it's less confusing, since the prototypes presented no longer match. What do you think?